### PR TITLE
[TR] PIM-5211: display the variant group edit form in less than 2 sec

### PR DESCRIPTION
**Pull request by jjanvier**
Tested on a catalog with 10k attributes. xdebug disabled.
NB: curl calls are slighty faster than real UI calls.

 Mean time of 25 curls call to `/enrich/variant-group/4/edit':
  * before patch: 10.068s (more than 10k requests to DB)
  * after patch: 1.602s (165 requests to DB)

Mean time of 25 curls call to `/enrich/group/5/edit':
  * before patch: 10.983s (more than 10k requests to DB)
  * after patch: 1.237s (160 request to DB)

| Q                 | A
| ----------------- | ---
| Specs             | Y
| Behats            | Y
| Blue CI           | running
| Changelog updated | Y
| Review and 2 GTM  |

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+use Pim\Bundle\CatalogBundle\Model\GroupInterface;
+use Pim\Bundle\CatalogBundle\Repository\AttributeRepositoryInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * Adds axes to the variant group form
+ *
+ * @author    Julien Janvier <filips@akeneo.com>
```
Comment (skeleton) [inline src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php]: filips ;)

Comment (solivier) [inline src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php]: filips :trollface: ?

File: src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class AddVariantGroupAxesSubscriber implements EventSubscriberInterface
+{
+    /** @var string */
+    protected $attributeClass;
+
+    /**
+     * AddVariantGroupAxesSubscriber constructor.
```
Comment (skeleton) [inline src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php]: useless

File: src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function preSetData(FormEvent $event)
+    {
+        $group = $event->getData();
+        $options = [
+            'label'    => 'Axis',
```
Comment (skeleton) [inline src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php]: remove static label, add a translation key

**Comment by solivier** [comment]
GTM after comments fixed

File: src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php (hunk 1 above)
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php]: did you changed your email address ?

File: src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    /**
+     * @param GroupInterface $group
+     *
+     * @return array
+     */
+    protected function getFormOptionsForEdition(GroupInterface $group)
+    {
+        $axesIds = array_map(
```
Comment (skeleton) [inline src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php]: IMO a foreach is more readable...

File: src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php (hunk 3 above)
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php]: no translations ?

**Comment by juliensnz** [comment]
GTM

File: src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php (hunk 1 above)
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php]: doen

File: src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php (hunk 2 above)
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php]: done

File: src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php (hunk 3 above)
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Form/Subscriber/AddVariantGroupAxesSubscriber.php]: this a translation key :) https://github.com/akeneo/pim-community-dev/blob/master/src/Pim/Bundle/EnrichBundle/Resources/translations/messages.en.yml#L602

**Pull request merged**